### PR TITLE
Cycle Sort in Go 

### LINKED ISSUE #2146: Cycle Sort
- [ ] C - @saloniankita
- [x] C++
- [x] C#
- [ ] CoffeeScript
- [x] Dart - @Rukmini-Meda
- [x] Go - @m-code12
- [x] Java
- [x] JS - @JanaSabuj
- [ ] Kotlin
- [ ] Php
- [x] Python
- [x] Readme
- [x] Ruby - @HardevKhandhar
- [ ] Rust
- [ ] TypeScript

### PR DESCRIPTION
**Pull request by m-code12**
Fixes #2146 

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [x] Input is taken dynamically.
- [x] Sample Input / Output is added at the end of file.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

#### Changes proposed in this pull request:
- Added Cycle Sort in Go

#### Languages Used:
- Go

#### Files Added:
- Cycle_Sort.go

> We're happy to help you get this ready -- don't be afraid to ask for help.

Thanks!

### CODE REVIEW

File: Cycle_Sort/Cycle_Sort.go [hunk 1]
```diff
@@ -0,0 +1,66 @@
+package main
+
+import (
+	"fmt"
+)
+
+func main() {
+	var x int
```
Comment (singh-shreya6) [inline Cycle_Sort/Cycle_Sort.go:8]: Incorrect indentation.

File: Cycle_Sort/Cycle_Sort.go [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+    fmt.Printf("Enter the size of the array : ")
+    fmt.Scan(&x)
+    fmt.Printf("Enter the elements of the array : ")
+    a := make([]int, x)
+    for i := 0; i < x; i++ {
+        fmt.Scan(&a[i])
+    }
+    arr := a    // A copy of the array 'a' is assigned to 'randomSlice' 
+	CycleSort(&arr)
```
Comment (singh-shreya6) [inline Cycle_Sort/Cycle_Sort.go:17]: Incorrect indentation

File: Cycle_Sort/Cycle_Sort.go [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+					pos++
+				}
+				(*arr)[pos], item = item, (*arr)[pos]
+				writes++
+			}
+		}
+	}
+
+	return writes
+}
```
Comment (singh-shreya6) [inline Cycle_Sort/Cycle_Sort.go:66]: Please add a blank line at end of file.

File: Cycle_Sort/Cycle_Sort.go [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+    for i := 0; i < x; i++ {
+        fmt.Scan(&a[i])
+    }
+    arr := a    // A copy of the array 'a' is assigned to 'randomSlice' 
+	CycleSort(&arr)
+	fmt.Printf("After sort: %v\n", arr)
+}
+
+func CycleSort(arr *[]int) int {
+	writes := 0
```
Comment (singh-shreya6) [inline Cycle_Sort/Cycle_Sort.go:22]: Looks like there are more than 4 spaces indentation in this function. Please check.

**Comment by m-code12** [comment]
@singh-shreya6 I have made all the requested changes.
Please review and merge.

**Pull request merged**